src/index.js

```diff
@@ -133,29 +134,32 @@ class Knob extends Component<KnobProps, KnobState> {
 
   // converts a value to an angle for the knob respecting the additional rotateDegrees prop
   convertValueToAngle(value: number): number {
-    const angle = (this.scale(value) + this.props.rotateDegrees) % 360;
+    const angle = (this.getScale()(value) + this.props.rotateDegrees) % 360;
     return this.normalizeAngle(angle);
   }
 
-
   // converts an angle to a value for the knob respecting the additional rotateDegrees prop
   convertAngleToValue(angle: number): number {
     const angle2 = angle - this.props.rotateDegrees;
-    return this.scale.invert(this.normalizeAngle(angle2))
+    return this.getScale().invert(this.normalizeAngle(angle2))
   }
 
-  setupScaling(min, max, clampMin, clampMax) {
-    // scaling of min/max to degree values needs to be swapped when turning the knob by 180 degree, to have the minimum values on the left side
-    if (this.props.rotateDegrees < 270 && this.props.rotateDegrees > 90) {
-      this.scale = scaleLinear()
-        .domain([min, max])
-        .range([clampMin, clampMax]);
-    } else {
-      this.scale = scaleLinear()
-        .domain([max, min])
-        .range([clampMax, clampMin]);
+  getScale() {
+    // Memoize scale so it's not recalculated everytime
+    if (!this.scaleProps || !(this.scaleProps.min === this.props.min && this.scaleProps.max === this.props.max && this.scaleProps.clampMin === this.props.clampMin && this.scaleProps.clampMax === this.props.clampMax)) {
+      if (this.props.rotateDegrees < 270 && this.props.rotateDegrees > 90) {
```

Comment:
One thing that "smells" bad is that we are excluding `props.rotateDegrees` from the memoization but it's used to generate the scale. I did not want to change the behavior without understanding this better but it might be a bug.